soh/soh/Enhancements/randomizer/randomizer_check_tracker.cpp

```diff
@@ -0,0 +1,789 @@
+    }
+
+    //Clean up last area
+    if (doDraw)
+        ImGui::TreePop();
+
+    ImGui::EndTable(); //Checks Lead-out
+    ImGui::EndTable(); //Quick Options Lead-out
+    EndFloatWindows();
+    if (doingCollapseOrExpand) {
+        optCollapseAll = false;
+        optExpandAll = false;
+    }
+}
+
+// Windowing stuff
+void BeginFloatWindows(std::string UniqueName, ImGuiWindowFlags flags) {
+    ImGuiWindowFlags windowFlags = flags;
+
+    if (windowFlags == 0) {
+        windowFlags |=
+            ImGuiWindowFlags_AlwaysVerticalScrollbar | ImGuiWindowFlags_NoFocusOnAppearing;
+    }
+
+    if (!CVar_GetS32("gCheckTrackerWindowType", 0)) {
+        ImGui::SetNextWindowViewport(ImGui::GetMainViewport()->ID);
+        windowFlags |= ImGuiWindowFlags_NoDocking | ImGuiWindowFlags_NoNav | ImGuiWindowFlags_NoTitleBar |
+                       ImGuiWindowFlags_NoScrollWithMouse | ImGuiWindowFlags_NoScrollbar;
+
+        if (!CVar_GetS32("gCheckTrackerHudEditMode", 0)) {
+            windowFlags |= ImGuiWindowFlags_NoInputs | ImGuiWindowFlags_NoMove;
+        }
+    }
+    ImGui::PushStyleColor(ImGuiCol_WindowBg, Color_Background);
+    ImGui::PushStyleColor(ImGuiCol_Border, ImVec4(0, 0, 0, 0));
+    ImGui::PushStyleVar(ImGuiStyleVar_WindowRounding, 4.0f);
+    ImGui::Begin(UniqueName.c_str(), nullptr, windowFlags);
+}
+void EndFloatWindows() {
+    ImGui::PopStyleVar();
+    ImGui::PopStyleColor();
+    ImGui::PopStyleColor();
+    ImGui::End();
+}
+
+
+void InitializeChecks() {
+    if (gPlayState == nullptr || gSaveContext.fileNum < 0 || gSaveContext.fileNum > 2)
+        return;
+
+    int count = 0;
+    for (auto& [rcCheck, rcObj] : RandomizerCheckObjects::GetAllRCObjects()) {
+        if (!RandomizerCheckObjects::IsVisibleInCheckTracker(rcObj))
+            continue;
+
+        checks.push_back(rcObj);
+        checkStatusMap.emplace(rcObj.rc, RCSHOW_UNCHECKED);
+        count++;
+        areaChecksTotal[rcObj.rcArea]++;
+    }
+
+    UpdateChecks();
+    UpdateOrdering(true);
+    doInitialize = false;
+    initialized = true;
+}
+
+void Teardown() {
+    initialized = false;
+    //delete checkStatus; //TODO may cause crashes due to race conditions during a draw cycle, but if you don't have it, then there's a memory leak when switching files
+    areasFullyChecked = 0;
+    checks.clear();
+    lastSaveCount = -1;
+    lastLocationChecked = RC_UNKNOWN_CHECK;
+    for (int i = 0; i < sizeof(areaChecksTotal); i++) {
+        areaChecksTotal[i] = 0;
+        areaChecksGotten[i] = 0;
+    }
+}
+
+bool ShouldUpdateChecks() {
+    // TODO eventually will need to be hooked into game elements rather than just save file
+    // TODO, saveCount isn't actually a thing. If not allowed to add it, will need to iterate over save bits and see if any changed 
+    // TODO, or enhance hooks, but that is a LUS change
+    //return lastSaveCount != gSaveContext.sohStats.saveCount;
+
+    return true;
+}
```

Comment:
Yes :( I couldn't find an obvious way to check to see if the game has been saved. If it's fine with you, I'll add the `saveCount` method. Very simple, low impact, good performance gain, and most importantly, I don't have to figure out how to do submodules :)

Reply:
not everyone uses autosave, the check tracker shouldn't rely on the game being saved to update

Reply:
Agreed! I've got a stub in place for Checked so that you eventually won't need to save to see it update, manual or auto regardless. This is also the reason ShouldUpdateChecks is a function, for all the possible hooks that could change the list of checked places; render functions, collected but unsaved checks (with or without reset), etc.

I've been going with the approach of "best is the enemy of the good" so far. I too want to see it come up by collection only, but that does increase the scope quite a lot, and might not be the task for a first time committer to SoH. I could try, but I make no promises I'm experienced enough to be up to the task :3

Reply:
I'd prefer the check tracker get merged with a known performance issue (assuming it's only an issue when the tracker is open) than only function when someone has autosave enabled 

In that case, the performance improvements could be their own PR, and anyone who feels confident enough in their abilities and motivated enough to work on it could implement them